client/login/controller.js

```diff
@@ -139,3 +139,13 @@ export function redirectDefaultLocale( context, next ) {
 		context.redirect( '/log-in' );
 	}
 }
+
+export function redirectJetpack( context, next ) {
+	const { isJetpack } = context.params;
+	const { redirect_to } = context.query;
+
+	if ( isJetpack !== 'jetpack' && includes( redirect_to, 'jetpack/connect' ) ) {
```

Comment:
I was wondering if we needed something similar to conditions used in #22488:

https://github.com/Automattic/wp-calypso/blob/29461f2b27917c63a488609e7b238e60630346b6/client/layout/masterbar/logged-out.jsx#L107-L108

But your more general approach here is probably better. May be nice to add leading slash: `/jetpack/connect`.

This made me wonder about SSO, which should presumably begin using the `log-in/jetpack` route. We shouldn't be hitting WordPress.com logins via SSO on mobile, should we?

Reply:
> May be nice to add leading slash: /jetpack/connect.

👍 

> This made me wonder about SSO, which should presumably begin using the log-in/jetpack route. We shouldn't be hitting WordPress.com logins via SSO on mobile, should we?

I agree it makes sense to brand the SSO flow login with Jetpack as well, as it's functionality is provided by Jetpack in the first place.

Reply:
> We shouldn't be hitting WordPress.com logins via SSO on mobile, should we?

No, not in the mobile apps. The only apps case we need to worry about is the jetpack.wordpress.com redirect to login when email matches a known account.